hugr-model/src/v0/text/parse.rs

```diff
@@ -0,0 +1,716 @@
+use bumpalo::Bump;
+use pest::{
+    iterators::{Pair, Pairs},
+    Parser, RuleType,
+};
+use thiserror::Error;
+
+use crate::v0::{
+    AliasDecl, FuncDecl, GlobalRef, LinkRef, LocalRef, MetaItem, Module, Node, NodeId, Operation,
+    Param, Port, Region, RegionId, RegionKind, Term, TermId,
+};
+
+mod pest_parser {
+    use pest_derive::Parser;
+
+    // NOTE: The pest derive macro generates a `Rule` enum. We do not want this to be
+    // part of the public API, and so we hide it within this private module.
+
+    #[derive(Parser)]
+    #[grammar = "v0/text/hugr.pest"]
+    pub struct HugrParser;
+}
+
+use pest_parser::{HugrParser, Rule};
+
+/// A parsed HUGR module.
+///
+/// This consists of the module itself, together with additional information that was
+/// extracted from the text format.
+#[derive(Debug, Clone)]
+pub struct ParsedModule<'a> {
+    /// The parsed module.
+    pub module: Module<'a>,
+    // TODO: Spans
```

Comment:
TODO here?

Reply:
Source spans will be useful, but aren't essential at the moment. So https://github.com/CQCL/hugr/issues/1550